### PR TITLE
adds Placement Configuration for rook-ceph agents and services

### PR DESCRIPTION
**Pull request by umangachapagain**
- updates CSV to change rook-ceph operator deployment
  enables nodeAffinity and toleration env var for rook agents
- updates CephCluster CR
  adds nodeAffinity, toleration, podAntiAffinity to all services

Signed-off-by: Umanga Chapagain <chapagainumanga@gmail.com>

### CODE REVIEW

**Comment by jarrpa** [comment]
Yeah, sorry, I didn't let my brain fully start up it seems.

File: pkg/controller/storagecluster/reconcile.go [hunk 1]
```diff
@@ -118,6 +119,86 @@ func newCephCluster(sc *ocsv1alpha1.StorageCluster) *rookCephv1.CephCluster {
...
 			},
+			Placement: rook.PlacementSpec{
+				"all": rook.Placement{
+					NodeAffinity: &corev1.NodeAffinity{
+						RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+							NodeSelectorTerms: []corev1.NodeSelectorTerm{
+								corev1.NodeSelectorTerm{
+									MatchExpressions: []corev1.NodeSelectorRequirement{
+										corev1.NodeSelectorRequirement{
+											Key:      "cluster.ocs.openshift.io/openshift-storage=",
```
[the remark this answers is not in the document]
Comment (umangachapagain) [reply pkg/controller/storagecluster/reconcile.go]: fixed.

[the remark this answers is not in the document]
Comment (umangachapagain) [reply pkg/controller/storagecluster/reconcile.go]: It is not. Removed.

File: pkg/controller/storagecluster/reconcile.go [hunk 2]
```diff
@@ -118,6 +119,86 @@ func newCephCluster(sc *ocsv1alpha1.StorageCluster) *rookCephv1.CephCluster {
...
+											Operator: metav1.LabelSelectorOpIn,
+											Values:   []string{"rook-ceph-mon"},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				"mgr": rook.Placement{
```
[the remark this answers is not in the document]
Comment (umangachapagain) [reply pkg/controller/storagecluster/reconcile.go]: Agreed. But, was considering a scenario if someone tried to scale up.

File: pkg/controller/storagecluster/reconcile.go [hunk 3]
```diff
@@ -118,6 +119,86 @@ func newCephCluster(sc *ocsv1alpha1.StorageCluster) *rookCephv1.CephCluster {
...
+											Operator: metav1.LabelSelectorOpIn,
+											Values:   []string{"rook-ceph-mgr"},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				"osd": rook.Placement{
```
[the remark this answers is not in the document]
Comment (umangachapagain) [reply pkg/controller/storagecluster/reconcile.go]: How does the Placement in StorageDeviceSet relate/propagate to OSD Placement?

StorageDeviceSet is something I never understood.

**Comment by umangachapagain** [comment]
>     * The CRs for object storage and file system should have the node affinity.
>
I do not see definitions for these resources in OCS Operator. @kshlm Where do I find these?

**Comment by Madhu-1** [comment]
> A couple areas of placement that are missing here:
> 
> * The CSI driver. Some of the pods (e.g. attacher) could have node affinity, though the driver pods would need to run everywhere that clients might need the storage. However, not sure we have a way to control the csi driver node affinity from Rook. @Madhu-1 ?

Missed this one, currently, we don't have node affinity for provisioner pod(which contains provisioner,snapshotter,liveness,  plugin and attacher).
> * The CRs for object storage and file system should have the node affinity.

**Pull request merged**